enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/upstream/strategies/LeaderOnlyStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.upstream.strategies;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.neo4j.causalclustering.discovery.RoleInfo;
+import org.neo4j.causalclustering.identity.MemberId;
+import org.neo4j.causalclustering.upstream.UpstreamDatabaseSelectionException;
+import org.neo4j.causalclustering.upstream.UpstreamDatabaseSelectionStrategy;
+import org.neo4j.helpers.Service;
+
+@Service.Implementation( UpstreamDatabaseSelectionStrategy.class )
+public class LeaderOnlyStrategy extends UpstreamDatabaseSelectionStrategy
```

Comment:
Do we want this to be available to production? If so do we want to document it?

Reply:
I just needed a thing like this, and don't know the answers to these questions. This is your domain, so, you tell me :)

Reply:
Perhaps @martinfurmanski has an opinion?

Reply:
@andrewkerr9000 We actually discussed this yesterday and came to the conclusion Yes. I'll make a card on our wall to do so.